src/SPC/builder/BuilderProvider.php

```diff
@@ -29,6 +30,10 @@ public static function makeBuilderByInput(InputInterface $input): BuilderBase
             'BSD' => new BSDBuilder($input->getOptions()),
             default => throw new WrongUsageException('Current OS "' . PHP_OS_FAMILY . '" is not supported yet'),
         };
+
+        // bind the builder to ExceptionHandler
+        ExceptionHandler::$bind_builder = self::$builder;
```

Comment:
I don't think we should be using static state for this. A method with a private member `$builders` would be better.